plugins/Marketplace/Plugins.php

```diff
@@ -343,4 +358,74 @@ private function addMissingRequirements($plugin)
 
         return $plugin;
     }
+
+    /**
+     * Find the cheapest shop variant, and if none is found specified, return the first variant.
+     *
+     * @param $plugin
+     * @return void
+     */
+    private function addPriceFrom(&$plugin): void
+    {
+        $variations = $plugin['shop']['variations'] ?? [];
+
+        if (!count($variations)) {
+            $plugin['priceFrom'] = null;
+            return;
+        }
+
+        $plugin['priceFrom'] = array_shift($variations); // use first as the default
+
+        foreach ($variations as $variation) {
+            if ($variation['cheapest'] ?? false) {
+                $plugin['priceFrom'] = $variation;
+                return;
+            }
+        }
+    }
+
+    /**
+     * If plugin is by Matomo, check if we have a specific image, otherwise fallback to default Matomo image.
+     * If plugin is not by Matomo, use generic preview image for now (until plugin categories are introduced).
+     *
+     * @param $plugin
+     * @return void
+     */
+    private function addPluginPreviewImage(&$plugin): void
+    {
+        $previewImage = 'generic-plugin';
+
+        if (in_array(strtolower($plugin['owner']), ['piwik', 'matomo-org'])) {
+            if (file_exists(sprintf('%s/images/previews/Matomo/%s.png', Manager::getPluginDirectory('Marketplace'), $plugin['name']))) {
+                $previewImage = 'Matomo/' . $plugin['name'];
+            } else {
+                $previewImage = 'matomo-plugin';
+            }
+        }
+
+        $plugin['previewImage'] = 'plugins/Marketplace/images/previews/' . $previewImage . '.png';
+    }
+
+    /**
+     * Add prettified number of downloads to plugin info to shorten large numbers to 1k or 1m format.
+     *
+     * @param $plugin
+     * @return void
+     */
+    private function prettifyNumberOfDownloads(&$plugin): void
+    {
+        $num = $nice = $plugin['numDownloads'] ?? 0;
+
+        if (($num >= 1000) && ($num < 100000)) {
+            $nice = round($num / 1000, 1, PHP_ROUND_HALF_DOWN) . 'k';
+        }
+        elseif (($num >= 100000) && ($num < 1000000)) {
+            $nice = floor($num / 100000) . 'k';
+        }
+        elseif ($num >= 1000000) {
+            $nice = floor($num / 1000000) . 'm';
+        }
```

Comment:
Could do, however how often are the marketplace related texts translated?
We'd need to have a confirmation from design that we can drop the decimal points as the CLDR notation seems to only support integers whereas here we have e.g. `16.3k` as the number. For lower numbers like `1.9k` if we floor'd it to 1k it might be too much of a drop and rounding to 2k is incorrectly adding to our downloads counts.

I'm 50/50 on this so happy to take guidance.

Reply:
Looked at this again and I'd leave it as is, if that's ok with you.